src/components/Layout/Header/NavBar/ChainMenu.tsx

```diff
@@ -51,19 +58,60 @@ type ChainMenuProps = BoxProps
 
 export const ChainMenu = (props: ChainMenuProps) => {
   const { state, load } = useWallet()
-  const { isLoading, supportedEvmChainIds, connectedEvmChainId, setEthNetwork } = useEvm()
+  const {
+    connectedEvmChainId,
+    getChainIdFromEthNetwork,
+    isLoading,
+    setEthNetwork,
+    supportedEvmChainIds,
+  } = useEvm()
   const chainAdapterManager = getChainAdapterManager()
   const translate = useTranslate()
 
-  const handleChainClick = async (chainId: ChainId) => {
-    try {
-      await (state.wallet as ETHWallet).ethSwitchChain?.(Number(chainId))
-      setEthNetwork(chainId)
-      load()
-    } catch (e) {
-      // TODO: Handle me after https://github.com/shapeshift/hdwallet/pull/551 is published
-    }
-  }
+  const handleChainClick = useCallback(
+    async (requestedEthNetwork: string) => {
+      try {
+        const requestedChainId = getChainIdFromEthNetwork(requestedEthNetwork)
+
+        if (!requestedChainId) {
+          throw new Error(`Unsupported EVM network: ${requestedEthNetwork}`)
+        }
+
+        const requestedChainChainAdapter = chainAdapterManager.get(requestedChainId) as unknown as
+          | EvmBaseAdapter<EvmChainId>
+          | undefined
+
+        if (!requestedChainChainAdapter) {
+          throw new Error(`No chain adapter found for: ${requestedChainId}`)
+        }
+
+        const requestedChainFeeAssetId = requestedChainChainAdapter.getFeeAssetId()
+        const requestedChainFeeAsset = selectAssetById(store.getState(), requestedChainFeeAssetId)
+
+        const requestedChainRpcUrl = requestedChainChainAdapter.getRpcUrl()
+        await (state.wallet as ETHWallet).ethSwitchChain?.({
+          chainId: utils.hexValue(Number(requestedEthNetwork)),
+          chainName: requestedChainChainAdapter.getDisplayName(),
+          nativeCurrency: {
+            name: requestedChainFeeAsset.name,
+            symbol: requestedChainFeeAsset.symbol,
+            decimals: 18,
```

Comment:
ser

Reply:
Actually disregard me, that's fine

Reply:
where have i heard that before?

Reply:
For future reference, that's *actually* fine: https://eips.ethereum.org/EIPS/eip-3085 - technically, non-18 decimals are possible, however, MetaMask itself type it as 18 and don't have explicit support for any other value, see:

https://github.com/MetaMask/metamask-docs/blob/b84cf77c15e098ea00ff64cbc0a64d1df1ea5944/docs/guide/rpc-api.md#parameters-3